test/integration/16-cluster-broker-web.js

```diff
@@ -0,0 +1,361 @@
+const HappnerCluster = require('../..');
+var Promise = require('bluebird');
+var expect = require('expect.js');
+
+var libDir = require('../_lib/lib-dir');
+var baseConfig = require('../_lib/base-config');
+var stopCluster = require('../_lib/stop-cluster');
+var users = require('../_lib/users');
+var testclient = require('../_lib/client');
+
+var clearMongoCollection = require('../_lib/clear-mongo-collection');
+//var log = require('why-is-node-running');
+describe(require('../_lib/test-helper').testName(__filename, 3), function() {
+
+  this.timeout(40000);
+
+  var servers = [];
+
+  function localInstanceConfig(seq, sync, dynamic) {
+    var config = baseConfig(seq, sync, true);
+    let brokerComponentPath = dynamic ? libDir + 'integration-10-broker-component-dynamic' : libDir + 'integration-09-broker-component';
+    config.cluster = config.cluster || {};
+    config.cluster.dependenciesSatisfiedDeferListen = true;
+    config.modules = {
+      'localComponent': {
+        path: libDir + 'integration-09-local-component'
+      },
+      'brokerComponent': {
+        path: brokerComponentPath
+      }
+    };
+    config.components = {
+      'localComponent': {
+        startMethod: 'start',
+        stopMethod: 'stop'
+      },
+      'brokerComponent': {
+        startMethod: 'start',
+        stopMethod: 'stop'
+      }
+    };
+    return config;
+  }
+
+  function remoteInstanceConfig(seq, sync) {
+    var config = baseConfig(seq, sync, true);
+    config.modules = {
+      'remoteComponent': {
+        path: libDir + 'integration-09-remote-component'
+      },
+      'remoteComponent1': {
+        path: libDir + 'integration-09-remote-component-1'
+      }
+    };
+    config.components = {
+      'remoteComponent': {
+        startMethod: 'start',
+        stopMethod: 'stop'
+      },
+      'remoteComponent1': {
+        startMethod: 'start',
+        stopMethod: 'stop',
+        web: {
+          routes: {
+            "testJSON": ["testJSON"],
+            "testJSONSticky": ["testJSONSticky"]
+          }
+        }
+      }
+    };
+    return config;
+  }
+
+  beforeEach('clear mongo collection', function(done) {
+    stopCluster(servers, function(e) {
+      if (e) return done(e);
+      servers = [];
+      clearMongoCollection('mongodb://localhost', 'happn-cluster', function() {
+        done();
+      });
+    });
+  });
+
+  afterEach('stop cluster', function(done) {
+    if (!servers) return done();
+    stopCluster(servers, function() {
+      clearMongoCollection('mongodb://localhost', 'happn-cluster', function() {
+        done();
+      });
+    });
+  });
+
+  function startInternal(id, clusterMin) {
+    return new Promise((resolve, reject) => {
```

Comment:
probably no need to wrap the promise returned by HappnerCluster.create().then() in another promise